src/Liuggio/StatsdClient/Factory/StatsdDataFactory.php

```diff
@@ -127,4 +134,37 @@ public function getEntityClass()
     {
         return $this->entityClass;
     }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPrefix($key)
+    {
+        $this->prefix = $key;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setSuffix($key)
+    {
+        $this->suffix = $key;
+
+        return $this;
+    }
+
+    /**
+     * @param $key
+     * @return string
+     */
+    public function getKeyMetric($key)
```

Comment:
Please make this function private, a builder pattern shall only expose the instance method and state mutators.